src/cgame/cg_view.cpp

```diff
@@ -1414,10 +1415,27 @@ static void CG_CalcColorGradingForPoint( vec3_t loc )
 		}
 	}
 
-	cg.refdef.gradingWeights[0] = 0.0f;
-	cg.refdef.gradingWeights[1] = haveGlobal ? ( 1.0f - totalWeight ) : ( selectedWeight[0] / totalWeight );
-	cg.refdef.gradingWeights[2] = totalWeight == 0.0f ? 0.0f : selectedWeight[1] / totalWeight;
-	cg.refdef.gradingWeights[3] = totalWeight == 0.0f ? 0.0f : selectedWeight[2] / totalWeight;
+	if ( totalWeight )
+	{
+		cg.refdef.gradingWeights[0] = 0.0f;
+		cg.refdef.gradingWeights[1] = selectedWeight[0];
+		cg.refdef.gradingWeights[2] = selectedWeight[1];
+		cg.refdef.gradingWeights[3] = selectedWeight[2];
+
+		cg.refdef.gradingWeights[0] /= totalWeight;
+		cg.refdef.gradingWeights[1] /= totalWeight;
+		cg.refdef.gradingWeights[2] /= totalWeight;
+		cg.refdef.gradingWeights[3] /= totalWeight;
+	}
+	else
+	{
+		Vector4Set( cg.refdef.gradingWeights, 0, 0, 0, 0 );
```

Comment:
This case should not exist because it would cause the screen to be rendered completely black.

Reply:
I don't know what that code does, but I have noticed the next test prevents it to be rendered black. So the original behavior (minus the division by zero) seem to have been conserved.

Reply:
I just rewrote the code a bit because I didn't want to do some hard-to-read `a ? b : c ? d : e` and while at it I rewrote it in a way it helps the compiler to vectorize it, but I expect the code to do the same as before, minus the division by zero.

Reply:
I managed to find a map where it can be tested, so I rewrote that code. https://github.com/Unvanquished/Unvanquished/pull/3213